src/main/java/elements/ElementFactoryImpl.java

```diff
@@ -0,0 +1,44 @@
+package elements;
+
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.openqa.selenium.WebElement;
+
+import java.lang.reflect.InvocationTargetException;
+
+import static java.text.MessageFormat.format;
+
+public class ElementFactoryImpl implements ElementFactory {
+    private final Logger logger = LogManager.getLogger("Element factory logger");
+
+    @Override
+    public <E extends CustomElement> E create(final Class<E> elementClass, final WebElement wrappedElement) {
+        try {
+            return findImplementationFor(elementClass)
+                    .getDeclaredConstructor(WebElement.class)
+                    .newInstance(wrappedElement);
+        } catch (InstantiationException e) {
+            logger.info("Error with exception: InstantiationException");
```

Comment:
Please don't log and throw exceptions at the same time, please check best practices for exceptions: https://stackify.com/best-practices-exceptions-java/